internal/pkg/service/stream/definition/repository/sink/sink_version.go

```diff
@@ -54,6 +54,15 @@ func (r *Repository) RollbackVersion(k key.SinkKey, now time.Time, by definition
 			// Prepare the new value
 			versionDescription := fmt.Sprintf(`Rollback to version "%d".`, targetVersion.Version.Number)
 			updated = deepcopy.Copy(*targetVersion).(definition.Sink)
+
+			// Preserve Disabled and Deleted states from the latest version.
+			// Without this it would be possible to bypass the Enable/Undelete event and avoid billing.
+			updated.Disabled = deepcopy.Copy(latestVersion.Disabled).(*definition.Disabled)
+			updated.Enabled = deepcopy.Copy(latestVersion.Enabled).(*definition.Enabled)
+			updated.Deleted = deepcopy.Copy(latestVersion.Deleted).(*definition.Deleted)
+			updated.Undeleted = deepcopy.Copy(latestVersion.Undeleted).(*definition.Undeleted)
```

Comment:
Can we do some method for deep copying the objects? I don't like the duplication. If we define that we want to preserve previous states of objects, it should be called as helper as the copy copies only certain fields, not whole objects fields.